### ISSUE TITLE
powerline fonts have mangled output

### ISSUE THREAD

**Issue by gvoysey** [nbedos/termtosvg#14]
on ubuntu 16.04, firefox 60.0.2, a recorded session shows mangled powerline fonts in my oh-too-blingy zsh config. 

compare static png: 
![image](https://user-images.githubusercontent.com/3641839/42235999-82ec592a-7ec7-11e8-89f2-b856f484b6c3.png)
attached SVG: 

[rendered.zip](https://github.com/nbedos/termtosvg/files/2160481/rendered.zip)

**Comment by immanuelfodor** [comment]
Same here. I use https://github.com/chris-marsh/pureline with Noto Mono on Manjaro Linux, and the special characters in the SVG are unrecognizable. Even when I try to provide the `--font="Noto Mono" ` parameter.

**Comment by nbedos** [comment]
Yeah, I noticed that with powerline fonts. I have no idea why the output is mangled though. All characters of a line in the SVG animation have the same vertical position and font size. What's the difference between powerline fonts and regular fonts?

**Comment by immanuelfodor** [comment]
Supporting more characters out of the box I think. However, asciinema
recordings are fine...


~ Sent from my mobile

**Comment by gvoysey** [comment]
>What's the difference between powerline fonts and regular fonts?  

I'm not sure in general. In my case, i'm using https://github.com/abertsch/Menlo-for-Powerline

**Comment by ylmrx** [comment]
I use the Nerd Font package and Tilix as a terminal emulator, it shows the same behavior for many of the fonts (within the terminal emulator and the browser). (these size problems come and go with the commit on the font repo side)
So, it's related to the font itself and pyte (the python library in charge of rendering a cool term window). hard to correct it in termtosvg.

@immanuelfodor : Maybe you unchecked the "allow webpage to specify font" in preferences > display > advanced (I had the very same problem). If you specify the font, it should be more like : "Noto Mono for Powerline" or something like this (I don't think Noto Mono will carry the "powerline" cool chars)

**Comment by immanuelfodor** [comment]
@ylmrx nope, it's untouched (default checked) in my FF.

I tried `termtosvg --font "Noto Mono for Powerline"` but it does not help. However, the Hack font works well, so it must be an issue with Noto Mono as you suggested in your last sentence. BTW does it mean that the user viewing the recording should have the font installed on his/her machine? Or is the font embedded into the SVG so the output is multiplatform and environment independent?

**Comment by ylmrx** [comment]
The SVG doesn't include the font. You need to have the specific font installed. 
It's not too bad as there's a failover font "monospace" (which your browser links to a default one), and it's platform-independent. However, this won't be a "powerline"-friendly (as it's perfectly non-standard)

**Comment by gvoysey** [comment]
>The SVG doesn't include the font. You need to have the specific font installed.

Which is interesting, because even when this is true the powerline output is still off.

**Comment by amcgregor** [comment]
> > The SVG doesn't include the font. You need to have the specific font installed.
> 
> Which is interesting, because even when this is true the powerline output is still off.

![SVG with embedded font.](http://s.webcore.io/e5301cc52f9c/2019-30-09--recording.svg)

Above is a hand-mangled clone of the `window_frame_js` template updated to tweak time alignment (which is entirely bonkers on Safari, but fine in Chrome), enable the close button (stops playback), an aborted attempt to make the maximize button do the HTML5 full-screen thing, and explicitly to embed Hasklig patched with _all_ of the icons, via [Nerdfont](https://nerdfonts.com).  Powerline's generally OK (a little painting outside the lines, but I can forgive that), but the lack of ligatures is killing me.  XP

**Comment by marcstreeter** [comment]
same here, I probably will just open regular terminal window and record from there until this is figured out.
